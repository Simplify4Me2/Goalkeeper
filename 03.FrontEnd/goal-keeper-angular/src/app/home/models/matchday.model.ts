import { Match } from "src/app/shared/models/match.model";

export interface Matchday {
    day: number;
    isOpeningMatchday: boolean;
    isClosingMatchday: boolean;
    matches: Match[];
}