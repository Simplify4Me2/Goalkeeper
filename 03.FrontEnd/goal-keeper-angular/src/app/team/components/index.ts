export * from './team-info/team-info.component';
export * from './team-players/team-players.component';
export * from './team-goals-stats/team-goals-stats.component';
export * from './team-cards-stats/team-cards-stats.component';
export * from './team-charts/team-charts.component';
export * from './player-card/player-card.component';