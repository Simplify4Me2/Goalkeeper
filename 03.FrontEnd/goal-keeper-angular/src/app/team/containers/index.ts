export * from './team.container';
export * from './matches/matches.container';
export * from './squad/squad.container';
export * from './stats/stats.container';