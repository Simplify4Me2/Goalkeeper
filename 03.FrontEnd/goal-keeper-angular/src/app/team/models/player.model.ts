export interface Player {
    id: number;
    firstName: string;
    lastName: string;
    nationality: string;
    shirtNumber: number;
    position: string;
}