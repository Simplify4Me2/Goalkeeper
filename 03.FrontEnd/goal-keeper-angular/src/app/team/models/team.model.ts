export interface Team {
    Name: string;
}