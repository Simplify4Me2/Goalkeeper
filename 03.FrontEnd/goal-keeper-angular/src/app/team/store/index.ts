export * from './team.reducer';
export * from './team.actions';
export * from './team.selectors';