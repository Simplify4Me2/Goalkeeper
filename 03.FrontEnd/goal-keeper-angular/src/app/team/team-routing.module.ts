import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

import { TeamContainerComponent, FixturesContainerComponent, SquadContainerComponent, StatsContainerComponent } from './containers';


const routes: Routes = [
    { 
        path: ':name', 
        component: TeamContainerComponent,
        children: [
            { path: 'fixtures', component: FixturesContainerComponent },
            { path: 'squad', component: SquadContainerComponent },
            { path: 'stats', component: StatsContainerComponent },
            { path: '', redirectTo: 'squad', pathMatch: 'full' },
            { path: '**', redirectTo: 'squad' },
          ],
     },
]

@NgModule({
    imports: [RouterModule.forChild(routes)],
    exports: [RouterModule]
})
export class TeamRoutingModule {}