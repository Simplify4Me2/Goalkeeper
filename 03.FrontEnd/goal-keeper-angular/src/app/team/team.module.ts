import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { TeamRoutingModule } from './team-routing.module';
import { TeamContainerComponent } from './containers/team.container';
import * as fromTeam from './store';
import { TeamEffects } from './store/team.effects';
import { TeamInfoComponent } from './components/team-info/team-info.component';
import { TeamPlayersComponent } from './components/team-players/team-players.component';
import { FixturesContainerComponent } from './containers/fixtures/fixtures.container';
import { PlayersContainerComponent } from './containers/players/players.container';
import { StatsContainerComponent } from './containers/stats/stats.container';

@NgModule({
  declarations: [TeamContainerComponent, TeamInfoComponent, TeamPlayersComponent, FixturesContainerComponent, PlayersContainerComponent, StatsContainerComponent],
  imports: [
    CommonModule,
    TeamRoutingModule,
    StoreModule.forFeature(fromTeam.teamFeatureKey, fromTeam.reducer),
    EffectsModule.forFeature([TeamEffects])
  ]
})
export class TeamModule { }
