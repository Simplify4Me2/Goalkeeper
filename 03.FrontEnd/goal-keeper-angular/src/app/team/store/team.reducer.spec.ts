import { teamReducer } from './team.reducer';
import { Team } from '../models';
import { getTeamSuccess } from './team.actions';

describe('TeamReducer', () => {
    it('an undefined action should return the default state', () => {
        const result = teamReducer(undefined, {} as any);

        expect(result).toEqual({ team: null });
        expect(result).toMatchSnapshot();
    });

    it('getTeamSuccess action should populate the state with a team', () => {
        const team: Team = { id: 3, name: 'FC De Kampioenen', players: [] };

        const result = teamReducer(undefined, getTeamSuccess({ team }));

        expect(result).toEqual({ team });
        expect(result).toMatchSnapshot();
    });
});