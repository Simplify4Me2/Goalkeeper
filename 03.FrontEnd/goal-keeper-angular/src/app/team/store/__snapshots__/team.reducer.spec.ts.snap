// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamReducer an undefined action should return the default state 1`] = `
Object {
  "team": null,
}
`;

exports[`TeamReducer getTeamSuccess action should populate the state with a team 1`] = `
Object {
  "team": Object {
    "id": 3,
    "name": "FC De Kampioenen",
    "players": Array [],
    "stadiumName": "DDT Ok Cars",
  },
}
`;
